invenio_oauthclient/contrib/keycloak/helpers.py

```diff
@@ -64,51 +102,13 @@ def get_user_info(remote, resp_token, fallback_to_endpoint=True, options=dict())
     :param options: A dictionary with additional options for `jwt.decode`
     """
     config_prefix = _generate_config_prefix(remote)
-    try:
-        # try to parse the "id_token" part of Keycloak's (JWT) response
-        token = resp_token["id_token"]
-        realm_url = current_app.config[f"{config_prefix}_REALM_URL"]
-        pubkey = _format_public_key(get_public_key(remote, realm_url))
-        alg = jwt.get_unverified_header(token)["alg"]
-
-        if not isinstance(options, dict):
-            options = {}
-
-        # consult the config whether to check the target audience
-        should_verify_aud = current_app.config.get(f"{config_prefix}_VERIFY_AUD", False)
-        expected_aud = current_app.config.get(f"{config_prefix}_AUD", None)
-
-        if should_verify_aud and (expected_aud is not None):
-            options.update({"verify_aud": True})
-        else:
-            options.update({"verify_aud": False})
-
-        # consult the config whether to check signature expiration
-        should_verify_expiration = current_app.config.get(
-            f"{config_prefix}_VERIFY_EXP", False
-        )
-
-        if should_verify_expiration:
-            options.update({"verify_exp": True})
-        else:
-            options.update({"verify_exp": False})
-
-        user_info = jwt.decode(
-            token, key=pubkey, algorithms=[alg], audience=expected_aud, options=options
-        )
-
-    except Exception as e:
-        if not fallback_to_endpoint:
```

Comment:
Fallback is always desired, if the fallback fails it will be raised in any case.

Reply:
i can't remember why i made that optional in the first place but i guess it was intended as a switch for either:
* saving network requests
* reducing communication with external services for security

given that the URL to be called is defined in the app config (and thus controlled by the admin) however, that should be okay as long as the communication uses valid SSL certificates